apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1628,6 +1651,19 @@ func (u *UniterAPI) NetworkInfo(args params.NetworkInfoParams) (params.NetworkIn
 	if err != nil {
 		return params.NetworkInfoResults{}, err
 	}
+	defaultSpaceAddress, spaceAddressErr := machine.PrivateAddress()
+	if args.RelationId != nil {
+		rel, err := u.st.Relation(*args.RelationId)
+		if err != nil {
+			return params.NetworkInfoResults{}, err
+		}
+		ru, err := u.getRelationUnit(canAccess, rel.Tag().String(), unitTag)
+		if err != nil {
+			return params.NetworkInfoResults{}, err
+		}
+		defaultSpaceAddress, spaceAddressErr = ru.SettingsAddress()
```

Comment:
I think we should rename SettingsAddress. It looked alarmingly like you were pulling the address out of the unit's relation settings. Not sure what's a better name.

Reply:
I've renamed to "IngressAddress" after discussing with John